spark_client/cli/service-account-registry.py

```diff
@@ -0,0 +1,189 @@
+        action="append",
+        type=str,
+        help="Config properties to be added to the service account.",
+    )
+
+    #  subparser for sa-conf-get
+    parser_account = subparsers.add_parser(str(Actions.GET_CONF.value))
+    parser_account.add_argument(
+        "--conf", action="append", type=str, help="Config property to retrieve."
+    )
+
+    #  subparser for sa-conf-del
+    parser_account = subparsers.add_parser(Actions.DELETE_CONF.value)
+
+    #  subparser for resources-primary-sa
+    parser_account = subparsers.add_parser(Actions.PRIMARY.value)
+
+    parser_account = subparsers.add_parser(Actions.LIST.value)
+
+    args = parser.parse_args()
+
+    logging.basicConfig(
+        format="%(asctime)s %(levelname)s %(message)s", level=args.log_level
+    )
+
+    kube_interface = KubeInterface(
+        defaults.kube_config, kubectl_cmd=defaults.kubectl_cmd
+    )
+
+    context = args.context or kube_interface.context_name
+
+    logging.info(f"Using K8s context: {context}")
+
+    registry = K8sServiceAccountRegistry(kube_interface.with_context(context))
+
+    if args.action == Actions.CREATE:
+
+        service_account = build_service_account_from_args(args)
+        service_account.extra_confs = (
+            PropertyFile.read(args.properties_file)
+            if args.properties_file is not None
+            else PropertyFile.empty()
+        ) + parse_conf_overrides(args.conf)
+
+        registry.create(service_account)
+
+    elif args.action == Actions.DELETE:
+
+        registry.delete(build_service_account_from_args(args).id)
+
+    elif args.action == Actions.UPDATE_CONF:
+
+        account_configuration = (
+            PropertyFile.read(args.properties_file)
+            if args.properties_file is not None
+            else PropertyFile.empty()
+        ) + parse_conf_overrides(args.conf)
+
+        registry.set_configurations(
+            build_service_account_from_args(args).id, account_configuration
+        )
+
+    elif args.action == Actions.GET_CONF:
+
+        input_service_account = build_service_account_from_args(args)
+
+        maybe_service_account = registry.get(input_service_account.id)
+
+        if maybe_service_account is None:
+            raise NoAccountFound(input_service_account.id)
+
+        maybe_service_account.configurations.log(print)
+
+    elif args.action == Actions.DELETE_CONF:
+
+        registry.set_configurations(
+            build_service_account_from_args(args).id, PropertyFile.empty()
+        )
+
+    elif args.action == Actions.PRIMARY:
+
+        maybe_service_account = registry.get_primary()
+
+        if maybe_service_account is None:
+            raise NoAccountFound()
+
+        maybe_service_account.configurations.log(print)
+
+    elif args.action == Actions.LIST:
+
+        for service_account in registry.all():
+            print(
+                f"{service_account.id}\t{service_account.primary}\t{json.dumps(service_account.extra_confs.props)}"
```

Comment:
maybe you can consider to use `str.expandtabs()` to have a better formatting.

Reply:
Definitely